contrib/intel/jenkins/tests.py

```diff
@@ -762,3 +762,92 @@ def execute_cmd(self, oneccl_test_gpu):
                         common.run_command(outputcmd)
 
+class DaosCartTest(Test):
+
+    def __init__(self, jobname, buildno, testname, core_prov, fabric,
+                 hosts, ofi_build_mode, user_env, util_prov=None):
+        super().__init__(jobname, buildno, testname, core_prov, fabric,
+                         hosts, ofi_build_mode, user_env, None, util_prov)
+
+        self.set_paths()
+        self.set_environment(core_prov,util_prov)
+        print(core_prov)
+        self.daos_nodes = ci_site_config.prov_node_map[core_prov]
+        print(self.daos_nodes)
+#        self.daos_nodes = node_dict[f'{core_prov}']
+#        print(self.daos_nodes)
+        self.cart_tests = {
+                 'corpc_one_node'            :       {'tags' :'cart,corpc,one_node', 'numservers':1, 'numclients':0},
+                 'corpc_two_node'            :       {'tags' :'cart,corpc,two_node', 'numservers':2, 'numclients':0},
+                 'ctl_one_node'              :       {'tags' :'cart,ctl,one_node', 'numservers':1, 'numclients':1},
+#                 'ghost_rank_rpc_one_node'   :       {'tags' :'cart,ghost_rank_rpc,one_node', 'numservers':1, 'numclients':0},
+                 'group_test'                :       {'tags' :'cart,group_test,one_node', 'numservers':1, 'numclients':0},
+                 'iv_one_node'               :       {'tags' :'cart,iv,one_node', 'numservers':1, 'numclients':1},
+                 'iv_two_node'               :       {'tags' :'cart,iv,two_node', 'numservers':2, 'numclients':1},
+                 'launcher_one_node'         :       {'tags' :'cart,no_pmix_launcher,one_node','numservers':1, 'numclients':1},
+                 'multictx_one_node'         :       {'tags' :'cart,no_pmix,one_node', 'numservers':1, 'numclients':0},
+                 'rpc_one_node'              :       {'tags' :'cart,rpc,one_node', 'numservers':1, 'numclients':1},
+                 'rpc_two_node'              :       {'tags' :'cart,rpc,two_node','numservers':2, 'numclients':1},
+                 'swim_notification'         :       {'tags' :'cart,rpc,swim_rank_eviction,one_node', 'numservers':1, 'numclients':1}
+        }
+
+
+    def set_paths(self):
+        self.ci_middlewares_path = f'{ci_site_config.ci_middlewares}'
+        self.daos_install_root = f'{self.ci_middlewares_path}/daos/install'
+        self.cart_test_scripts = f'{self.daos_install_root}/lib/daos/TESTING/ftest'
+        self.mpipath = f'{ci_site_config.daos_mpi}/bin'
+        self.pathlist = [f'{self.daos_install_root}/bin/', self.cart_test_scripts, self.mpipath, \
+                       f'{self.daos_install_root}/lib/daos/TESTING/tests']
+        self.daos_prereq = f'{self.daos_install_root}/prereq'
+        common.run_command(['rm','-rf', f'{self.daos_prereq}/debug/ofi'])
+        common.run_command(['ln', '-sfn', self.libfab_installpath, f'{self.daos_prereq}/debug/ofi'])
+
+    def set_environment(self, core_prov, util_prov):
```

Comment:
does slurm inherit the environment from the head node?
When I was setting environment variables for the MPI tests I had to do something like lines 283-295 in tests.py.
That way the command would be separated with ';' and would set my environment variables for me once the subprocess was running on the node.

Reply:
Currently we don't have slurm configured on the daos cluster since its only being internally used. 
we have dedicated nodes for CI.

Reply:
Thanks for checking